jsoncpp pkgconfig file reports wrong libs for static builds since 1.9.6 update

### Description / Steps to reproduce the issue

When building an application which uses jsoncpp statically and linking using the output of `pkgconfig --static --libs jsoncpp`, linking will fail. This started happening since jsoncpp was updated to version 1.9.6. I am running into this issue with my game OceanPop, available [here](https://github.com/sharkwouter/oceanpop). Below is a smaller example that should reproduce it.

main.cpp:
```
#include <string>
#include <json/json.h>
#include <iostream>

int main(void)
{
  Json::Value data;
  Json::Reader reader;
  std::string text = "{\"message\":\"hello json\"}";
  
  reader.parse(text, data);
  std::cout << data.get("message", "Could not load json message") << "\n";
  
  return 0;
}
```
CMakeLists.txt:
```
cmake_minimum_required(VERSION 3.10)

project(hello_json)

set(CMAKE_CXX_FLAGS "-static ${CMAKE_CXX_FLAGS}")

include(FindPkgConfig)
pkg_search_module(JSONCPP Required jsoncpp)

include_directories(${JSONCPP_INCLUDE_DIRS})
add_executable(${PROJECT_NAME} main.cpp)
target_link_libraries(${PROJECT_NAME} ${JSONCPP_STATIC_LIBRARIES})
```

Put these in a directory, then run the following commands there:
```
mkdir build && cd build
cmake -G "MinGW Makefiles" ..
cmake --build .
```

### Expected behavior

As with jsoncpp version 1.9.5, I was expecting the build to link correctly and result in a working binary.

### Actual behavior

As seen in the failure of [this GitHub workflow](https://github.com/sharkwouter/oceanpop/actions/runs/11912612163/job/33196581768), building with jsoncpp version 1.9.6 results in the following error:
```
 [100%] Linking CXX executable oceanpop.exe
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -ljsoncpp: No such file or directory
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -ljsoncpp: No such file or directory
collect2.exe: error: ld returned 1 exit status
mingw32-make[2]: *** [CMakeFiles\oceanpop.dir\build.make:517: oceanpop.exe] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:86: CMakeFiles/oceanpop.dir/all] Error 2
mingw32-make: *** [Makefile:135: all] Error 2
```
This same code build just fine before the update.

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-20348

### Are you willing to submit a PR?

I have contributed before, but I don't know if I'll have time to fix this myself.

--- a/mingw-w64-jsoncpp/0001-static-lib-name.patch
+++ b/mingw-w64-jsoncpp/0001-static-lib-name.patch
@@ -1,0 +1,14 @@
+Append suffix to static library name only for MSVC.
+
+diff -urN jsoncpp-1.9.6/src/lib_json/CMakeLists.txt.orig jsoncpp-1.9.6/src/lib_json/CMakeLists.txt
+--- jsoncpp-1.9.6/src/lib_json/CMakeLists.txt.orig	2024-09-10 05:18:29.000000000 +0200
++++ jsoncpp-1.9.6/src/lib_json/CMakeLists.txt	2024-11-19 14:07:18.639721600 +0100
+@@ -143,7 +143,7 @@
+ 
+     # avoid name clashes on windows as the shared import lib is also named jsoncpp.lib
+     if(NOT DEFINED STATIC_SUFFIX AND BUILD_SHARED_LIBS)
+-        if (WIN32)
++        if (MSVC)
+             set(STATIC_SUFFIX "_static")
+         else()
+             set(STATIC_SUFFIX "")

--- a/mingw-w64-jsoncpp/PKGBUILD
+++ b/mingw-w64-jsoncpp/PKGBUILD
@@ -4,7 +4,7 @@ _realname=jsoncpp
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=1.9.6
-pkgrel=2
+pkgrel=3
 pkgdesc="A C++ library for interacting with JSON (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
@@ -17,8 +17,24 @@ depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs")
 makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
              "${MINGW_PACKAGE_PREFIX}-cmake"
              "${MINGW_PACKAGE_PREFIX}-ninja")
-source=("${_realname}-${pkgver}.tar.gz"::"https://github.com/open-source-parsers/jsoncpp/archive/${pkgver}.tar.gz")
-sha256sums=('f93b6dd7ce796b13d02c108bc9f79812245a82e577581c4c9aabe57075c90ea2')
+source=("${_realname}-${pkgver}.tar.gz"::"https://github.com/open-source-parsers/jsoncpp/archive/${pkgver}.tar.gz"
+        "0001-static-lib-name.patch")
+sha256sums=('f93b6dd7ce796b13d02c108bc9f79812245a82e577581c4c9aabe57075c90ea2'
+            '692d7ae7f102644aaf7a41f1f9eafd7b3eca45febd6bde8675afdabca85adb4f')
+
+apply_patch_with_msg() {
+  for _patch in "$@"
+  do
+    msg2 "Applying ${_patch}"
+    patch -Nbp1 -i "${srcdir}/${_patch}"
+  done
+}
+
+prepare() {
+  cd "${srcdir}/${_realname}-${pkgver}"
+  apply_patch_with_msg \
+    0001-static-lib-name.patch
+}
 
 build() {
   declare -a extra_config

